### PR TITLE
JS-4 Executable for client and server

### PR DESCRIPTION
**Pull request by sagersmith8**
### Description
- Rework all code to rely on streamed resources
- Create executable jar for client
- Create msi installer for client

### Note
- Didn't include installer for server, not sure how we want to group those together

### Testing
- [ ] Client still executes correctly through ide (run Main)
- [ ] Client jar now executes as expected. (`./gradlew clean build`, run the jar from `client/build/libs`)
- [ ] Install the client installable, run the exe, verify it works. (`./gradlew javapackager`, click on client/build/distributions/bundle/Lame Game Client.msi`, allow it permissions to install, in the start menu type in `Lame Game Client`, verify it runs)

### Please Review:
@JReistad

### CODE REVIEW

File: client/build.gradle [hunk 1]
```diff
@@ -85,3 +83,60 @@ dependencies {
...
+        )
+    }
+}
+
+// always call copyJarsToLib when building jars:
+jar.dependsOn copyDependencies
+
+def embeddedJRE = true
+
+task javapackager(type: Exec, dependsOn: [assemble]) {
```
Comment (sagersmith8) [inline client/build.gradle:119]: May either want a ticket to copy this to the server, or have them packaged together top level with options how which one to run? Wasn't sure, your call @JReistad

File: client/build.gradle [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
-    mainClassName = 'com.github.jsappz.lamegame.game.Main'
 }
 
 repositories {
     mavenCentral()
 }
 
 dependencies {
+    // https://mvnrepository.com/artifact/commons-io/commons-io
+    compile group: 'commons-io', name: 'commons-io', version: '2.6'
```
Comment (sagersmith8) [inline client/build.gradle:11]: Probably  a follow up to add this to dependencies.gradle instead @JReistad

File: client/build.gradle (hunk 1 above)
Comment (sagersmith8) [reply client/build.gradle:119]: Another potential follow up is some sort of integration test setup between the client and the server. Docker might be a good option depending on how we plan to deploy the server. Your Call @JReistad

File: client/src/main/java/com/github/jsappz/lamegame/engine/graph/Texture.java [hunk 3]
```diff
@@ -20,8 +24,13 @@ public Texture(String fileName) throws Exception {
             IntBuffer w = stack.mallocInt(1);
             IntBuffer h = stack.mallocInt(1);
             IntBuffer channels = stack.mallocInt(1);
-
-            buf = stbi_load(fileName, w, h, channels, 4);
+            try (InputStream is = getClass().getClassLoader().getResourceAsStream(fileName)) {
```
Comment (sagersmith8) [inline client/src/main/java/com/github/jsappz/lamegame/engine/graph/Texture.java:27]: Note this code is going to be pretty expensive. It gets the resource input stream, reads all the bytes into a byte array, then copies all the bytes from a byte array into a byte buffer. I didn't see another good option, but might be worth a follow up I will let @JReistad decide

File: client/build.gradle (hunk 2 above)
Comment (JReistad) [reply client/build.gradle:11]: will write up a ticket for it

File: client/src/main/java/com/github/jsappz/lamegame/engine/graph/OBJLoader.java [hunk 4]
```diff
@@ -11,46 +14,49 @@
...
-                default:
-                    // Ignore other lines
-                    break;
+        try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream(file);
+             InputStreamReader isr = new InputStreamReader(is);
+             BufferedReader reader = new BufferedReader(isr)) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] tokens = line.split("\\s+");
+                switch (tokens[0]) {
```
Comment (JReistad) [inline client/src/main/java/com/github/jsappz/lamegame/engine/graph/OBJLoader.java:28]: an example of what a line in this file looks like would be helpful

File: client/src/main/java/com/github/jsappz/lamegame/engine/graph/Texture.java (hunk 3 above)
Comment (JReistad) [reply client/src/main/java/com/github/jsappz/lamegame/engine/graph/Texture.java:27]: will add todo for looking into this

File: client/build.gradle (hunk 1 above)
Comment (JReistad) [reply client/build.gradle:119]: 👍 thanks for calling out. will probably let this one stew a while until we figure out what seems fitting. Will start a long term backlog. 
We'll definitely want some integration tests between client and server at some point in time, but seems early at this point.

**Comment by JReistad** [comment]
Verified:
- [ ] Client still executes correctly through ide (run Main)
 - [ ] Client jar now executes as expected. (./gradlew clean build, run the jar from client/build/libs)
- [ ]  Install the client installable, run the exe, verify it works. (./gradlew javapackager, click on client/build/distributions/bundle/Lame Game Client.msi, allow it permissions to install, in the start menu type in Lame Game Client`, verify it runs)

**Pull request merged**